pkg/components/azure-arc/component_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azurearc_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/kinvolk/lokomotive/pkg/components/azure-arc"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+func TestEmptyConfig(t *testing.T) {
+	c := azurearc.NewConfig()
+
+	config := `component "azure-arc-onboarding" {}`
+
+	body, diagnostics := util.GetComponentBody(config, azurearc.Name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Error getting component body: %v", diagnostics)
+	}
+
+	if diagnostics := c.LoadConfig(body, &hcl.EvalContext{}); !diagnostics.HasErrors() {
+		t.Fatal("Empty config should return errors as all the fields are required fields.")
+	}
+}
+
+func TestConfigSuccessWithArbitraryValues(t *testing.T) {
+	c := azurearc.NewConfig()
+
+	config := `
+component azure-arc-onboarding {
+  application_client_id="test-application-id"
+  tenant_id="test-tenant-id"
```

Comment:
We should perhaps test each property individually.

Reply:
seems a lot of repetition. Might I suggest we add a test for number of errors in diagnostics ?

We have 5 fields, none of them optional. If we have a test with all fields empty, we should get 5 errors in diagnostics.

Reply:
Repetition is fine in tests many times, if it benefits the readability, I wouldn't be scared of that. Sure, if we would be able to skip the HCL parsing part, then we could just generate a valid config and make test function only zero single field and then expect an error.

Reply:
added a new test that covers everything. Let me know if that's fine. I'll delete the original redundant tests.

Reply:
@invidian ping

Reply:
Yeah, new tests looks fine. Maybe except the comments I left.